source/parser/parser_expressions.cpp

```diff
@@ -1222,16 +1222,22 @@ void Parser::Parse_Num_Factor (EXPRESS& Express,int *Terms)
                         // JN2007: Image map dimensions:
                         CASE (PIGMENT_ID_TOKEN)
```

Comment:
Currently this only works for `pigment`, but I think we also want it to work with `density`, so I suggest to replace the above with `CASE2 (PIGMENT_ID_TOKEN, DENSITY_ID_TOKEN)`.